pkg/kubelet/config/config_test.go

```diff
@@ -85,6 +89,14 @@ func CreatePodUpdate(op kubetypes.PodOperation, source string, pods ...*v1.Pod)
 	return kubetypes.PodUpdate{Pods: pods, Op: op, Source: source}
 }
 
+func createPodConfigTesterByChannel(mode PodConfigNotificationMode, channelName string) (chan<- interface{}, <-chan kubetypes.PodUpdate, *PodConfig) {
```

Comment:
could you explain a little bit why this is necessary?  since createPodConfigTester() does the same thing with channel named "TestSource"

Reply:
The API server restore get pushed to a channel for "api".

Reply:
maybe i missed something, correct me if i my understanding is wrong, does this test need a channel named "api" specifically?  or any channel name is fine, since the test own the channel, push and get update from it, thanks

Reply:
correct. The channel needs to be called "api" specifically. The function that is there will only receive on `Source: 'test'`.

Reply:
when i looked at the code further, i do think the channel named with `test` should work unless your actual testing prove it is not.
```
updates <- kubetypes.PodUpdate{Pods: pods, Op: kubetypes.RESTORE, Source: kubetypes.ApiserverSource}
```
the value of 'Source' at this line does not matter (is it?),  eventually all updates will merged to 'ch',   
only need to make sure it matches with source name below which you already did:
```
expectPodUpdate(t, ch, CreatePodUpdate(kubetypes.RESTORE, kubetypes.ApiserverSource, pod))
```

the code below should work:
```
channel, ch, config := createPodConfigTester(PodConfigNotificationIncremental)      
if err := config.Restore(tmpDir, channel); err != nil {     
    t.Fatalf("Restore returned error: %v", err)     
}     
expectPodUpdate(t, ch, CreatePodUpdate(kubetypes.RESTORE, kubetypes.ApiserverSource, pod))    
```

i could be wrong if actual test prove it,  my intention here: we don't have to add new function unless it is really 
necessary, thanks.

Reply:
ok, just realize that merge() will actually modify Source in PodUpdate based on channel name,   
the code below works fine, i tested it locally:
```
   channel, ch, config := createPodConfigTester(PodConfigNotificationIncremental)
   if err := config.Restore(tmpDir, channel); err != nil {
       t.Fatalf("Restore returned error: %v", err)
   }
   expectPodUpdate(t, ch, CreatePodUpdate(kubetypes.RESTORE, TestSource, pod))
```

Reply:
thanks for the review and tip... I updated the PR with that change.